core/src/Platforms/iOS/TokenCacheAccessor.cs

```diff
@@ -32,90 +32,248 @@
 using Microsoft.Identity.Core;
 using Microsoft.Identity.Core.Cache;
 using System.Collections.ObjectModel;
+using Microsoft.Identity.Core.Helpers;
 
 namespace Microsoft.Identity.Core
 {
     internal class TokenCacheAccessor : ITokenCacheAccessor
     {
-        private const string AccessTokenServiceId = "com.microsoft.identity.client.accessToken";
-        private const string RefreshTokenServiceId = "com.microsoft.identity.client.refreshToken";
-        private const string IdTokenServiceId = "com.microsoft.identity.client.idToken";
-        private const string AccountServiceId = "com.microsoft.identity.client.Account";
+        public const string CacheKeyDelimiter = "-";
+
+        static Dictionary<string, int> AuthorityTypeToAttrType = new Dictionary<string, int>()
+        {
+            {AuthorityType.AAD.ToString(), 1001},
+            {AuthorityType.MSA.ToString(), 1002},
+            {AuthorityType.MSSTS.ToString(), 1003},
+            {AuthorityType.OTHER.ToString(), 1004},
+        };
+
+        enum CredentialAttrType
+        {
+            AccessToken = 2001,
+            RefreshToken = 2002,
+            IdToken = 2003,
+            Password = 2004
+        }
 
         private const bool _defaultSyncSetting = false;
         private const SecAccessible _defaultAccessiblityPolicy = SecAccessible.AfterFirstUnlockThisDeviceOnly;
 
+        private readonly string DefaultKeychainGroup = "com.microsoft.adalcache";
+        private readonly string TeamIdKey = "teamIDHint";
+
+        private string keychainGroup;
+
+        private string GetBundleId()
+        {
+            return NSBundle.MainBundle.BundleIdentifier;
+        }
+   
+        public void SetSecurityGroup(string securityGroup)
```

Comment:
Not sure what this does, but I'd have 2 methods - one taking a string security group and throwing an ArgumentNull Exception if it is null or empty, and one taking 0 args and doing the GetBundleId() thing. Otherwise, it does too much.

Reply:
this is part of Existing Api to Msal iOS , implementation in .net just reflects it,
discussion of final API in .NET is still in progress.

Reply:
Sure, just saying this method does too much and it would make the code clearer if there 2 methods.

Reply:
pls. take in the code style discussion

---
In reply to: [199553595](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1108#discussion_r199553595) [](ancestors = 199553595)